src/test.js

```diff
@@ -507,7 +501,7 @@ Test.prototype = {
         ...test.hooks('before'),
 
         function () {
-          test.preserveTestEnvironment();
+          test.testEnvironment = extend({}, test.module.testEnvironment, false, true);
```

Comment:
Maybe it's not worth using `extend` here as I don't want only own properties from the module's testEnvironment. I added a new boolean arg to switch it to all properties.

Reply:
This essentially offsets the introduction of `Object.create()` between module and parent module, which previously used extend but with inheritence that means you miss out on props. Is that right?

Reply:
Yes, that's right